bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/FileFormat.java

```diff
@@ -84,6 +87,16 @@ List<ElementAtZoom<ImageData>> loadFromByteStream(int fileZoom, int targetZoom)
 		}
 	}
 
+	static abstract class DynamicImageFileFormat extends FileFormat {
+
+		abstract ImageData[] loadFromByteStream(int targetZoom);
+
+		@Override
+		List<ElementAtZoom<ImageData>> loadFromByteStream(int fileZoom, int targetZoom) {
+			return Arrays.stream(loadFromByteStream(targetZoom)).map(d -> new ElementAtZoom<>(d, fileZoom)).toList();
```

Comment:
```suggestion
			return Arrays.stream(loadFromByteStream(100 * targetZoom / fileZoom)).map(d -> new ElementAtZoom<>(d, targetZoom)).toList();
```

Reply:
The issue here is that `fileZoom` will usually be 100 for SVGs (as that's what we assume for them). This is because just rendering the SVG at original scale, it will produce a 100% image. With the mechanisms in Platform UI, you could also specify, for example, a 2x version of the SVG icon, but that does not really make sense. But to still ensure that it works, the `fileZoom` has to be considered when rendering the SVG, as a 200%-scaled SVG with a target zoom of 150 needs to be rendered at 100 * targetZoom / fileZoom = 100 * 150 / 200.

Reply:
I changed it according to your suggestion. Thank you :)